DoctrineMongoDBBundle.php

```diff
@@ -54,56 +49,21 @@ public function getContainerExtension()
 
     public function boot()
     {
-        // Register an autoloader for proxies to avoid issues when unserializing them
-        // when the ODM is used.
-        if (! $this->container->hasParameter('doctrine_mongodb.odm.proxy_namespace')) {
-            return;
-        }
-
-        $namespace = $this->container->getParameter('doctrine_mongodb.odm.proxy_namespace');
-        $dir       = $this->container->getParameter('doctrine_mongodb.odm.proxy_dir');
-        // See https://github.com/symfony/symfony/pull/3419 for usage of
-        // references
-        $container =& $this->container;
-
-        $this->autoloader = static function ($class) use ($namespace, $dir, &$container) {
-            if (strpos($class, $namespace) !== 0) {
-                return;
-            }
-
-            $fileName = str_replace('\\', '', substr($class, strlen($namespace) +1));
-            $file     = $dir . DIRECTORY_SEPARATOR . $fileName . '.php';
-
-            if (! is_file($file) && $container->getParameter('doctrine_mongodb.odm.auto_generate_proxy_classes')) {
-                $registry = $container->get('doctrine_mongodb');
-
-                // Tries to auto-generate the proxy file
-                foreach ($registry->getManagers() as $dm) {
-                    if (! $dm->getConfiguration()->getAutoGenerateProxyClasses()) {
-                        continue;
-                    }
+        /** @var ManagerRegistry $registry */
+        $registry = $this->container->get('doctrine_mongodb');
 
-                    $originalClassName = $dm->getClassNameResolver()->getRealClass($class);
-                    $classes           = $dm->getMetadataFactory()->getAllMetadata();
-
-                    foreach ($classes as $classMetadata) {
-                        if ($classMetadata->name !== $originalClassName) {
-                            continue;
-                        }
-
-                        $dm->getProxyFactory()->generateProxyClasses([$classMetadata]);
-                    }
-                }
+        $this->registerAutoloader($registry->getManager());
```

Comment:
What if there's more than one manager?

Reply:
I was thinking the same, but the proxy namespace is shared among document managers, so I would think the autoloader would be the same for all managers. I can take a look to see whether there would be any differences.

Reply:
Ah indeed, I thought that each manager can have its own namespace defined 👍

Reply:
Took another look: since we use the same values of `proxy_dir` and `auto_generate_proxy_classes` for all document managers, checking only the default manager will work as expected for the current configuration. If people decide to override the configuration for a single document manager, they'll have to ensure autoloading for said manager still works as expected.